app/models/spree/order.rb

```diff
@@ -149,11 +149,13 @@ def states
     }
 
     scope :sort_by_billing_address_name_asc, -> {
-      joins(:bill_address).order("spree_addresses.lastname ASC, spree_addresses.firstname ASC")
+      references(:bill_address)
+        .order("spree_addresses.lastname ASC, spree_addresses.firstname ASC")
```

Comment:
This happens when sorting on the normal orders page. It's a custom search scope and Ransack doesn't add any magic. The original query has `includes(:bill_address)` already which we could repeat here but I kept it minimal and just used `references`.

Reply:
I think it would be worth a code comment here explaining why the `references` needs to be here. (Of course, the specs will be the ultimate protection).